Automattic/wp-calypso
# Domains: Allow limiting the suggested TLDs by query string

#### Proposed Changes

We'll be migrating the TLD landing pages to start using Calypso instead of the old PHP pages.  So, we need to allow limiting the search results by TLD. This PR just uses the `tld` query parameter to set the TLDs requested from the domain suggestions endpoint.

#### Testing Instructions

Visit a page that uses the DomainStep component and add the `tld=` parameter to the query string. For example:

http://calypso.localhost:3000/start/domain/domain-only?new=foo+bar&search=yes&tld=art

Then make sure that all the suggestions use the specified TLD.

Also test for several TLDs using a comma-delimited list such as:

http://calypso.localhost:3000/start/domain/domain-only?new=foo+bar&search=yes&tld=art,wine,pizza

Make sure that the suggestions have all of those TLDs.

Also, visit the page without the `tld=` query parameter and make sure that all available TLDs are suggested.

Also make sure that using `&tld` and `&tld=` do not limit the selection.

Make sure that an invalid TLD in does not cause a problem. For example:

http://calypso.localhost:3000/start/domain/domain-only?new=foo+bar&search=yes&tld=foo

or

http://calypso.localhost:3000/start/domain/domain-only?new=foo+bar&search=yes&tld=---

This PR also removes the no-longer used TRUENAME promo coupons in order to simplify the TLD limiting as well.
